### ISSUE TITLE
Docker Desktop removes MobyLinuxVM from Hyper-V when Docker Desktop starts

### ISSUE THREAD

**Issue by roberts1000** [docker/for-win#3788]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior

Expect Docker Desktop for Windows to start without problems after starting the host.

### Actual behavior

When I start my machine in the morning to begin work, I first sign into my VPN.  Once the network is working on my host, I start Docker Desktop.  Docker Desktop **always** throws the following error:

```
Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Inconsistent parameters PolicyStore PersistentStore and Dhcp Enabled

   at Docker.Core.Pipe.NamedPipeClient.Send(String action, Object[] parameters) in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeClient.cs:line 36
   at Docker.Actions.DoStart(SynchronizationContext syncCtx, Boolean showWelcomeWindow, Boolean executeAfterStartCleanup) in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 92
   at Docker.Actions.<>c__DisplayClass19_0.<Start>b__0() in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 74
   at Docker.WPF.TaskQueue.<>c__DisplayClass19_0.<.ctor>b__1() in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.WPF\TaskQueue.cs:line 59
```

### Information

Docker Desktop appears to delete the MobyLinuxVM when Docker Desktop starts.  MobyLinuxVM shows up in Hyper-V before I start Docker Desktop.  It gets deleted a couple seconds after starting Docker Desktop. Once it's deleted, I can _sometimes_ get Docker to recreate the VM, but I have to reboot the machine and restart Docker multiple times.

  - Windows Version: 1803 (OS Build 17134.706)
  - Docker for Windows Version: 2.0.0.3 (31259) Build 8858db3

### Steps to reproduce the behavior

Assuming I'm at a state where Docker Desktop is working...

  1. Right click on the Docker Desktop tray icon and select `Quit Docker Desktop`.
  1. Once Docker Desktop closes, shutdown the host Windows 10 machine.
  1. Turn on the machine the next day.
  1. Connect to my VPN.
  1. Open Hyper-V to verify MobyLinxVM is listed.
  1. Start Docker Desktop for Windows.
  1. Wait about 10 seconds.
  1. Watch MobyLinuxVM get yanked from Hyper-V.
  1. The Docker Desktop tray icon becomes red and the error dialog (mentioned above) gets opened.
  
Failed attempts to fix:

   1. Try to restart Docker Desktop.  The tray icon will actually turn white, but MobyLinuxVM isn't recreated.
   1. Try to reinstall Docker Desktop.  It reinstalls, but MobyLinuxVM isn't recreated.

### Workaround to Recreate MobyLinuxVM

I have identified a workaround that lets me recreate the MobyLinuxVM. Maybe there's some info here that will help explain why Docker can't recreate its VM.

1. _If you've completed this step before, there's no need to do it again._ Right click on the Docker Desktop system tray icon and select the `Settings` menu option.  If Docker Desktop isn't running, start it. It's ok if it fails to fully start.
    1. Select the `General` tab.
    1. Make sure `Start Docker Desktop when you login in` in unchecked.  
    1. Close the `Settings` dialog.
1. Right click on the Docker Desktop system tray icon again.
    1. Select `Quit Docker Desktop`.
    1. Wait for Docker Desktop to fully stop.  If it won't stop, open the Windows Task Manager and kill the process.
1. Open PowerShell as an Admin. 
    1. Execute `cd C:\Program Files\Docker\Docker\resources`
    1. Execute `Stop-Service com.docker.service`
    1. Execute `.\MobyLinux.ps1 -Destroy -KeepVolume`.  Even though MobyLinuxVM isn't present in Hyper-V, the `MobyLinuxVM.vhdx` storage file is probably still alive (for me it's at `C:\Users\Public\Public Documents\Hyper-V\Virtual hard disks`). This file contains all your images and containers.  If it gets deleted, you'll lose all of that information.  `-KeepVolume` preserves this file so Docker Desktop can reuse it when it recreates the VM in Hyper-V.  Without it, Docker Desktop will delete it and try to fresh start.
    1. Execute `.\MobyLinux.ps1 -Create` again.
    1. If the command fails with this message: `Inconsistent Parameters PolicyStore PersistentStore and Dhcp Enabled`, fix it by doing this (this issue is reported [here](https://github.com/docker/for-win/issues/2050)):
        1. In PowerShell (or the Windows search box), execute `ncpa.cpl` to open the `Network Connections` dialog.
        1. In the `Network Connections` window, find `vEthernet (DockerNAT)`.  It probably has a red X and says "Not Connected". Right click on it and select `Properties`.  You'll need to enter your admin account to proceed. A dialog will open:
            1. In the `This connection uses the following items` list of the dialog, uncheck `Cisco AnyConnect Network Access Manager Filter Driver`.
            1. In the same list, select `Internet Protocol Version 4 (TCP/IPv4)`.  Don't uncheck it. Just press the  the `Properties` button to open another dialog.
                1. Make sure `Obtain an IP address automatically` and `Use DNS server address automatically` are check.  Press `OK`.
                1. Close the Dialog.
            1. The red X on the `vEthernet (DockerNAT)` icon will go away and the text will switch to "Identifying...".  Wait for it to change to "Unidentified Network" (it takes a few seconds).
        1. Close the `Network Connections` UI.
        1. Now execute `.\MobyLinux.ps1 -Create` in PowerShell again.
        1. If it fails with this message: `Logon failure: the user has not been granted the requested logon type at this computer`, fix it by doing this:
            1. Reboot your machine.
            1. Login to your desktop.
            1. Reopen PowerShell as admin.
               1. Execute `cd C:\Program Files\Docker\Docker\resources`
               1. Execute `.\MobyLinux.ps1 -Create`
1. Once the `.\MobyLinux.ps1 -Create` command succeeds, Docker should have created the MobyLinuxVM.  Open Hyper-V Manager to confirm.
1. Now attempt to start Docker Desktop.

**Comment by roberts1000** [comment]
I hit this same issue a couple times on a Windows 1709 machine as well, but Docker seems to be able to recreate the MobyLinuxVM on its own most of the time.  Docker got stuck once and it couldn't recreate the VM.  After about 5 reboots and restarts of Docker, I gave up and worked on something else.  Interestingly, when I tried again an hour later, it repaired itself without problem.  Maybe there's a virus scan, or some other process that runs when the machine starts and it doesn't play well with Docker.
